### DISCUSSION TITLE
Extending the JoltC coverage

### DISCUSSION THREAD

**Discussion by jmiskovic** [amerkoleci/JoltPhysicsSharp#27] [Q&A]
Hi! Thanks for creating the JoltC. I'm using it as a base to interface the Jolt from Lua, which means I need to create Lua bindings on top of C level bindings.

I noticed some parts of Jolt is not covered by JoltC, so I started extending with calls I needed. I've created maybe ~30 more calls so far, covering different constraints, mass, motion damping. My plan is to cover few more constraints, and then move towards meshes, heightfield shapes and collision callbacks. Working code is available [here](https://github.com/jmiskovic/lovr/tree/jolt/src/lib/joltc).

Would you accept the contributions of these extra APIs? Note that I'm only extending the JoltC part, not touching the sharp side.

Bonus question:
All the C API constructors have suffix `Create`, it's unfortunate that there is a conflict with a few dozen classes that contain the `Create()` method themselves. For example, I needed to support `JPH_HingeConstraintSettings` which has both the constructor and the `Create()` method. How would you handle this?

**Comment by amerkoleci** [comment]
Absolutely, feel free to add what you need and I help out.

Thanks!

**Reply by amerkoleci** [reply]
Some bindings where added (https://github.com/amerkoleci/JoltPhysicsSharp/commit/cb19b4a5bd45d2987652ad39fba7347c27c2d456) feel free to add others, make sure to use JPH_Bool32 instead of bool.

Thanks
